components/ParameterForm.vue

```diff
@@ -235,10 +236,11 @@ const invalidFields = computed(() => {
   return invalids;
 });
 
+// Since some defaults depend on the values of other fields, this function should not be used to initialize form values.
 const defaultValue = (param: Parameter) => {
   if (param.updateNumericFrom) {
     const dependedOnParamId = param.updateNumericFrom.parameterId;
-    return param.updateNumericFrom.values[formData.value![dependedOnParamId]].default.toString();
+    return param.updateNumericFrom.values[formData.value[dependedOnParamId]]?.default?.toString();
```

Comment:
Sorry, I can't really tell what's prompted this change or what it's actually doing, and the comment doesn't really help. 

Is it using the default of the parameter this parameter depends on, if it's not already set..? So that would be happen if we're trying to get default hospital capacity, but we haven't initialised country yet?

This logic of this line is very tangled - I think splitting it up onto a couple of lines would help a lot to show what the intermediate values you're using are, and why.

Reply:
I think this change was prompted by the situation with Thailand -- something like, the country value was being defaulted to Thailand, but since that wasn't a country, there was no corresponding updateNumericFrom values? -- and so "Checking whether a field value is already at its default" (something done at render time) threw an error preventing the page from loading. So the addition of the question marks covers us (allows the form to still load) when we can't find any default value. But maybe we want the dramatic failure of a 500 page to alert us to the problem.

Happy to refactor to be clearer, anyhow.

Reply:
```ts
if (param.updateNumericFrom) {
    const dependedOnParamId = param.updateNumericFrom.parameterId;
    const dependedOnValue = formData.value[dependedOnParamId];
    const dependentDefaultValue = param.updateNumericFrom.values[dependedOnValue]?.default;
    return dependentDefaultValue?.toString();
```

Reply:
Thanks for explanation - yeah, that makes sense I think. That situation shouldn't really happen, but I guess it's good to be somewhat resilient!